### ISSUE TITLE
Navigation Block: Aria-current attribute to indicate the current page

### ISSUE THREAD

**Issue by TeBenachi** [WordPress/gutenberg#35567]
### Description

Aria-current can inform the screen reader users which page within the navigation  is currently active. It can also benefit the sighted users when the current page is visually indicated using css or preferably theme.json.

### Step-by-step reproduction instructions

1. Turn on the screen reader
2. Navigate to the currently open page in the navigation menu using the Tab key.

### Environment info

Firefox on macOS, VoiceOver

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by TeBenachi** [comment]
Thank you @talldan 

Would you consider adding this ticket as well? #35122
The screen reader doesn’t read the name of the navigation links on Firefox on macOS. It reads “Empty navigation” instead. 

Chrome and Safari work without any issues. 


![Firefox](https://user-images.githubusercontent.com/30734209/137110597-77b0fb3d-9761-4cb4-b0a1-2733031e2220.gif)

**Comment by talldan** [comment]
@TeBenachi 👍 Done

**Comment by TeBenachi** [comment]
Thank you!!
